sympy/solvers/ode/ode.py

```diff
@@ -599,7 +599,7 @@ def recur_len(l):
             raise NotImplementedError
         else:
             if match['is_linear'] == True:
-                if match['no_of_equation'] > 3:
+                if all([str in match and match[str] for str in ['is_constant', 'is_homogeneous']]):
```

Comment:
How about
```
if match.get('is_constant', False) and match.get('is_homogeneous', False):
```

Reply:
Yes this is better. Thanks for the suggestion. I will update this line in the next commit.